libraries/botframework-streaming/src/interfaces/INodeSocket.ts

```diff
@@ -6,13 +6,151 @@
+    on(event: "data", listener: (data: INodeBuffer) => void): this;
+    on(event: "end", listener: () => void): this;
+    on(event: "error", listener: (err: Error) => void): this;
+
+    once(event: "close", listener: () => void): this;
+    once(event: "data", listener: (chunk: any) => void): this;
+    once(event: "end", listener: () => void): this;
+    once(event: "readable", listener: () => void): this;
+    once(event: "error", listener: (err: Error) => void): this;
+    once(event: string | symbol, listener: (...args: any[]) => void): this;
+
+    prependListener(event: "close", listener: () => void): this;
+    prependListener(event: "data", listener: (chunk: any) => void): this;
+    prependListener(event: "end", listener: () => void): this;
+    prependListener(event: "readable", listener: () => void): this;
+    prependListener(event: "error", listener: (err: Error) => void): this;
+    prependListener(event: string | symbol, listener: (...args: any[]) => void): this;
+
+    prependOnceListener(event: "close", listener: () => void): this;
+    prependOnceListener(event: "data", listener: (chunk: any) => void): this;
+    prependOnceListener(event: "end", listener: () => void): this;
+    prependOnceListener(event: "readable", listener: () => void): this;
+    prependOnceListener(event: "error", listener: (err: Error) => void): this;
+    prependOnceListener(event: string | symbol, listener: (...args: any[]) => void): this;
+
+    rawListeners(event: string | symbol): Function[];
+
+    removeAllListeners(event?: string | symbol): this;
+    removeListener(event: string | symbol, listener: (...args: any[]) => void): this;
+
+    [Symbol.asyncIterator](): AsyncIterableIterator<any>;
+    
+    _read(size: number): void;
+    read(size?: number): any;
+
+    setDefaultEncoding(encoding: string): this;
+    setEncoding(encoding: string): this;
+    setMaxListeners(n: number): this;
+    getMaxListeners(): number;
+    setTimeout(timeout: number, callback?: () => void): this;
+    setKeepAlive(enable?: boolean, initialDelay?: number): this;
+    setNoDelay(noDelay?: boolean): this;
+    
+    _write(chunk: any, encoding: string, callback: (error?: Error | null) => void): void;
+    _writev?(chunks: Array<{ chunk: any, encoding: string }>, callback: (error?: Error | null) => void): void;
+
+    write(buffer: ValidBuffer, cb?: (err?: Error) => void): boolean; 
+    write(str: string, encoding?: string, cb?: Function): boolean; 
+    write(buffer: ValidBuffer): boolean; 
+    write(str: string, cb?: Function): boolean; 
+    write(str: string, encoding?: string, fd?: string): boolean; 
+    write(data: any, encoding?: string, callback?: Function): void; 
+    write(chunk: any, cb?: (error: Error | null | undefined) => void): boolean;
+    write(chunk: any, encoding?: string, cb?: (error: Error | null | undefined) => void): boolean;
+
+    ref(): any;
+    unref(): any;
+}
+
+interface AddressInfo {
```

Comment:
It didn't complain that these two interfaces weren't explicitly exported?

Reply:
No, as I was just only using them within the file for `INodeSocket`